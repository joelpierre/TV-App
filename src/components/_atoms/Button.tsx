import classNames from 'classnames';
import React from 'react';

import styles from './Button.module.scss';

interface IButtonProps {
  className?: string;
}

const Button: React.FunctionComponent<IButtonProps> = ({
  className,
  children
}) => {
  return (
    <button data-testid="Button" className={classNames(styles.Button, className)}>
      {children}
    </button>
  );
};

export default Button;
