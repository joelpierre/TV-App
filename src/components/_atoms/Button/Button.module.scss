.Button {
  width: auto;
  padding: spacing() spacing(2);
  background-color: $color-brand;
  appearance: none;
  cursor: pointer;
  border: none;
  color: $color-tint-alpha;
  border-radius: (spacing(0.5) / 2);
  font-weight: bold;
  transition: background-color .4s;

  @include hover-focus {
    background-color: $color-alpha;
  }
}
