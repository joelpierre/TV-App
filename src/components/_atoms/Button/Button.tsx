import classNames from 'classnames';
import React from 'react';

import styles from './Button.module.scss';

interface IButtonProps {
  className?: string;
  onClick?: (event: React.MouseEvent) => void;
}

const Button: React.FunctionComponent<IButtonProps> = ({
  className,
  children,
  onClick
}) => {
  return (
    <button onClick={onClick} data-testid="Button" className={classNames(styles.Button, className)}>
      {children}
    </button>
  );
};

export default Button;
