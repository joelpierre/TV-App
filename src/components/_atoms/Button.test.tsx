import React from 'react';
import { render, RenderResult } from 'test-utils';

import Button from './Button';

const children = 'This is a test';

const defaultProps = {
  children
};

const setup = (props = {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props };
  return render(<Button {...setupProps} />);
};

describe('<Button/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should container children', () => {
    expect(wrapper.container).toHaveTextContent(children);
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
