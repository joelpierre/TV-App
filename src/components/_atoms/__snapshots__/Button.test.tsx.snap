// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button"
    data-testid="Button"
  >
    This is a test
  </button>
</DocumentFragment>
`;
